Altoverview2

Mingling94 opened the pull request:
Even more abstraction and less repetitious code than #24. :+1: or :-1:?

seveibar commented:
Hey Matt, 

Great PR! It looks like the indentation in `footer.haml` got messed up though (it was switched from spaces to tabs). Could you change that back?

leej42 commented:
Looks Great

seveibar commented:
Great job Matt! Looks great!

Mingling94 committed:
Alternate way of calling api from overview plugin

Mingling94 committed:
Alternate with further abstraction

DavidVorick commented:
I think this one is the best. Do you want me to merge it and close #24? (Luke is in agreement)

Mingling94 committed:
Updated tutorial to reflect abstraction changes

This is detailed in extra section called "Bonus: Abstracting the Extra
Mile"

Also made use of github markdown syntax "Diff" and added more comments

Mingling94 commented:
@DavidVorick @lukechampine
The Overview's javascript remains the same for the most part with the last commit, though I made a few corrections. 

Peruse the tutorial, specifically the last section, and merge if satisfied ^_^

DavidVorick commented on doc/plugin-tutorial.md:
````diff
@@ -136,11 +136,11 @@ unique id's to update each of them separately later in JS.
-    └── button.png
+```diff
+ Sia-UI/app/plugins/Overview/
++├── index.html
````
this is magic

DavidVorick commented on doc/plugin-tutorial.md:
```diff
@@ -259,14 +258,14 @@ Quite a lot to take in without review, but that's how styling webpages goes.
+ │   ├── button.png
++│   └── roboto-condensed-min.css
++└── css/
++    └── overview.css
```
dark magic

DavidVorick commented:
looks good, tutorial still doesn't cover jsdoc
